### PR TITLE
Doc and code cleanup

### PR DESCRIPTION
**Pull request by junlee1991**

### CODE REVIEW

**Review by bakano98 — COMMENTED** [review]
Image 0 is still wrong -- the visibility is supposed to follow this
image

Please fix and repush ASAP.

File: src/main/java/seedu/address/logic/commands/UndoCommand.java [hunk 1]
```diff
@@ -12,22 +12,18 @@
...
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireAllNonNull(model, undoRedoStack);
 
         if (!undoRedoStack.canUndo()) {
             throw new CommandException(MESSAGE_USAGE_FAILURE);
+        } else {
+            undoRedoStack.popUndo().undo(model);
+            return new CommandResult(MESSAGE_USAGE_SUCCESS);
         }
```
Comment (lawwm) [inline src/main/java/seedu/address/logic/commands/UndoCommand.java:26]: Removing the check for SwitchCommand breaks undo, redo for that command.

File: src/main/java/seedu/address/logic/commands/StackUndoRedo.java [hunk 2]
```diff
@@ -46,17 +63,15 @@ public void push(Command command) {
...
 
     /**
-     * Pops and returns the next {@code UndoableCommand} to be redone in the stack.
+     * Pops and returns the next {@code RedoableCommand} to be redone in the stack.
      */
     public RedoableCommand popRedo() {
         RedoableCommand toRedo = redoStack.pop();
         undoStack.push(toRedo);
-        logger.info(this.toString());
         return toRedo;
```
Comment (lawwm) [inline src/main/java/seedu/address/logic/commands/StackUndoRedo.java:75]: Why remove logging the state of the stack?

File: src/main/java/seedu/address/logic/commands/StackUndoRedo.java [hunk 3]
```diff
@@ -29,15 +25,36 @@ public StackUndoRedo() {
...
 
-        undoStack.add((RedoableCommand) command);
+    /**
+     * Returns true if the command is either an undo command or a redo command.
+     */
+    public boolean isNotUndoRedoCommand(Command command) {
+        if (!(command instanceof UndoCommand) && !(command instanceof RedoCommand)) {
+            return true;
+        }
+        return false;
```
Comment (lawwm) [inline src/main/java/seedu/address/logic/commands/StackUndoRedo.java:57]: Why refactor this into a different method?

File: src/main/java/seedu/address/logic/commands/RedoableCommand.java [hunk 4]
```diff
@@ -36,7 +33,7 @@ private void saveAddressBookSnapshot(Model model) {
     protected final void undo(Model model) {
         requireAllNonNull(model, this.previousAddressBook);
 
-        model.setSwappedAddressBook(this.isSwapped, this.previousAddressBook, this.previousArchiveBook);
+        model.setAddressBook(this.previousAddressBook);
```
Comment (lawwm) [inline src/main/java/seedu/address/logic/commands/RedoableCommand.java:37]: By removing saving the state of both addressbook and archivebook, undo redo is broken.

**Review by lawwm — COMMENTED** [review]
Changes to code added more bugs.

File: src/main/java/seedu/address/logic/commands/UndoCommand.java [hunk 5]
```diff
@@ -12,22 +12,18 @@
 
     public static final String COMMAND_WORD = "undo";
     public static final String MESSAGE_USAGE_SUCCESS = "Undo success!";
-    public static final String MESSAGE_USAGE_FAILURE = "No more commands to undo!";
+    public static final String MESSAGE_USAGE_FAILURE = "Unable to perform undo!";
```
Comment (bakano98) [inline src/main/java/seedu/address/logic/commands/UndoCommand.java]: Don't change error messages. Not allowed

File: src/main/java/seedu/address/logic/commands/StackUndoRedo.java (hunk 3 above)
Comment (bakano98) [reply src/main/java/seedu/address/logic/commands/StackUndoRedo.java:57]: ^, and even though it's good to use SLAP here, you would not need two methods. Just one is enough.

File: src/main/java/seedu/address/logic/commands/UndoCommand.java (hunk 1 above)
Comment (bakano98) [reply src/main/java/seedu/address/logic/commands/UndoCommand.java]: Yep, can confirm this causes the bug. To reproduce:
1. Switch
2. Undo
3. States of `archivedAddressBook.json` and `addressbook.json` gets messed up.

**Comment by bakano98** [comment]
Please check the latest comments and update accordingly

**Comment by lawwm** [comment]
Btw, I realised that class diagrams were used when they should have been object diagrams. Will be changing.

**Review by fantablack — COMMENTED** [review]
There is an extra copy of LogicClassDiagram which I believe as been erroneously added as it is exactly the same as the existing one.

**Comment by lawwm** [comment]
1. Change some diagrams to object diagrams.
2. Expanded upon original class diagram.
3. Change sequence diagram to use brackets for conditions and align more closely with actual code.
4. Remove redundant logic class diagram.

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -299,89 +299,90 @@ The following activity diagram shows how a `Person` with `Status` and `Module` i
...
 
-* `VersionedAddressBook#commit()` — Saves the current address book state in its history.
-* `VersionedAddressBook#undo()` — Restores the previous address book state from its history.
-* `VersionedAddressBook#redo()` — Restores a previously undone address book state from its history.
+- The Undo stack serves to store a "history" of the actions they have performed.
+- The Redo stack is a collection of their actions that lead up to initial condition at which they started performing the undo.
 
-These operations are exposed in the `Model` interface as `Model#commitAddressBook()`, `Model#undoAddressBook()` and `Model#redoAddressBook()` respectively.
+They central concept is to store a stack of commands that essentially functions as a history-list of the commands. Essentially, we leverage on the stack's data structure of the which is a linear data structure that is based on the principle of Last In First Out (LIFO). Based on the implementation described above, the undo stack is populated by pushing a user's action in the application to the undo stack.
```
Comment (Btaykb) [inline docs/DeveloperGuide.md:311]: Possible typo? Should be "The central concept..." instead

Comment (lawwm) [reply docs/DeveloperGuide.md]: Good catch.

**Comment by bakano98** [comment]
I updated `UndoRedo1.png` to be more representative of the code.

Note that you can have multiple stop points (in case anyone is confused), which has been clarified in one of the tutorials.

**Comment by lawwm** [comment]
LGTM

**Pull request merged**